src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -610,28 +708,35 @@ static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
 
 static void glb_rr_connectivity_changed(grpc_exec_ctx *exec_ctx, void *arg,
                                         grpc_error *error) {
-  /* If shutdown or error free the arg. Rely on the rest of the code to set the
-   * right grpclb status. */
-  rr_connectivity_data *rr_conn_data = arg;
-  glb_lb_policy *glb_policy = rr_conn_data->glb_policy;
-  gpr_mu_lock(&glb_policy->mu);
+  rr_connectivity_data *rr_connectivity = arg;
+  glb_lb_policy *glb_policy = rr_connectivity->glb_policy;
 
-  if (rr_conn_data->state != GRPC_CHANNEL_SHUTDOWN &&
-      !glb_policy->shutting_down) {
-    /* RR not shutting down. Mimic the RR's policy state */
-    grpc_connectivity_state_set(exec_ctx, &glb_policy->state_tracker,
-                                rr_conn_data->state, GRPC_ERROR_REF(error),
-                                "rr_connectivity_cb");
-    /* resubscribe. Reuse the "rr_connectivity_cb" weak ref. */
+  gpr_mu_lock(&glb_policy->mu);
+  const bool shutting_down = glb_policy->shutting_down;
+  grpc_lb_policy *maybe_unref = NULL;
+  GRPC_ERROR_REF(error);
```

Comment:
This follows the pattern of similar functions such as `pf_connectivity_changed` (https://github.com/grpc/grpc/blob/master/src/core/ext/lb_policy/pick_first/pick_first.c#L259), whereby a ref is kept for the duration of the function/callback. This is due to the ownership rules in iomgr/error.h, which don't guarantee that a callback owns the error instance it receives. The ref is taken to ensure the `error` doesn't disappear mid-function.